src/docs/content/controlled.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Controlled
+description: MeltUI components are uncontrolled by default, but offer the ability to be controlled.
+---
+
+<script>
+    import { Callout } from '$docs/components'
+</script>
+
+<Callout type="warning">
+
+Controlled components are an optional advanced feature and should only be used when necessary. If
+you're unsure if you _need_ to use controlled components, you probably don't.
+
+</Callout>
+
+## Controlled vs. Uncontrolled
+
+There are various definitions for controlled/uncontrolled components, but in the context of MeltUI,
+uncontrolled means that the state and stores of a component are created and managed by us.
+Controlled means that the user (you) can create and manage the state and stores.
+
+All components are uncontrolled by default, and in most cases, this is more than enough. However,
+our goal is to provide as much flexibility as possible, so we offer a few ways to give you more
+control over the state & behavior of the components.
+
+## Controlled Usage
+
+For the following examples, we'll use the `Dialog` builder, but the same concepts also apply to
+other builders. Be sure to reference the API docs for your builder to see what controlled options
+are available.
+
+### Bring Your Own Store (BYOS)
+
+Without you having to do anything, we create an `open` store within the Dialog builder, which
+updates when a trigger or close element is clicked/key-downed. We do return this store for you to
+use, but let's say you wanted to define your own `open` store in another file and share it with
+other components in your app that might need to track or update the open state of the dialog.
+
+You could accomplish this using context or some other hacky solution, but things may become
+complicated quickly. Instead, we provide a way to pass your own `open` store to the builder.
+
+It's as simple as passing your own `open` store to the `createDialog` builder.
```

Comment:
```suggestion
By default, we provide an `open` store from the `Dialog` builder that will have its state 
updated whenever a `trigger` or `close` element is clicked/key-downed. 

If you wanted to define your own `open` store so that its state could be shared and 
updated by other parts of your app, then we offer a way for you to supply your own.

It's as simple as passing your own `open` store to the `createDialog` builder.
```